directives/visitor.go

```diff
@@ -26,5 +27,5 @@ type ResolverInterceptor interface {
 
 // Validator directive which executes before anything is resolved, allowing the request to be rejected.
 type Validator interface {
```

Comment:
Thinking that the type (& function) may need a rename before a release is made. I believe there are 3 different times where parts of the schema could be “visited” (are there more?):

* When the schema is parsed (during parsing? After successful?)
* While handling a request _before_ executing any resolvers ie validation currently, part of query complexity scoring soon
* While executing resolvers for a request ie as resolver middleware, currently applied in the interceptor types

Unsure how each would ideally be named for clarity, and for consistency with other GraphQL implementations

Reply:
Yes, you are right. I think that the middleware makes sense to be called interceptors as they are definitely not visitors. A visitor implies visiting all fields in the schema or the current executable document. So I would call the existing directives interceptors and I would allow for a pluggable visitor plugin which has access to the field directives. These directives, however, would be just decorative and would not have any callback themselves. The author of the visitor should decide whether and how to use them when it makes sense.